src/flag_gems/utils/libentry.py

```diff
@@ -14,47 +16,110 @@ def __init__(
         while not isinstance(fn, triton.runtime.JITFunction):
             fn = fn.fn
         self.jit_function: triton.runtime.JITFunction = fn
-        self.kernel_arg_indices = []
+        self.spec_indices = []
+        self.dns_indices = []
         for p in self.jit_function.params:
             if not p.is_constexpr:
-                self.kernel_arg_indices.append(p.num)
+                if p.do_not_specialize:
+                    self.dns_indices.append(p.num)
+                else:
+                    self.spec_indices.append(p.num)
```

Comment:
Isn't a bit of list comprehension is more favorable than loops?

Reply:
done